File: leetcode/longestSubstringWithoutRepeatition.cpp
// leetcode - https://leetcode.com/problems/longest-substring-without-repeating-characters
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int maxLength=0,start=0;
        map<char,int> hash;
        for(int i=0;i<s.size();i++)
        {
            if(hash.find(s[i])!=hash.end() && start<=hash[s[i]])
                start=hash[s[i]]+1;
            else
            {
                maxLength=max(maxLength,i-start+1);
            }
            hash[s[i]]=i;
        }
        return maxLength;
    }
};

File: leetcode/longestSubstringWithoutRepeatition.py
#leetcode https://leetcode.com/problems/longest-substring-without-repeating-characters
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        map={}
        max_length=start=0
        for i in range(len(s)):
            if s[i] in map and start<=map[s[i]]:
                start=map[s[i]]+1
            else:
                max_length=max(max_length,i-start+1)
            map[s[i]]=i
        return max_length
